sips/sip-019/sip-019-token-metadata-update-notifications.md

```diff
@@ -0,0 +1,278 @@
+```
+
+| Key                   | Value                                                                  |
+|-----------------------|------------------------------------------------------------------------|
+| `notification`        | The string `"token-metadata-update"`                                   |
+| `payload.token-class` | The string `"nft"`                                                     |
+| `payload.token-ids`   | (optional) A list with the uint token ids that need to be refreshed    |
+| `payload.contract-id` | The contract id (principal) of the contract that declared the tokens   |
+
+If a notification does not contain a value for `payload.token-ids`, it means it is requesting an
+update for all tokens.
+
+## Considerations for metadata indexers
+
+For a token metadata update notification to be considered valid by metadata indexers, it must meet
+the following requirements:
+
+1. Its payload structure should be correct whether it is updating a [FT](#fungible-tokens) or a
+   [NFT](#non-fungible-tokens).
+1. The STX address that initiated the contract call which emits the notification should match the
+   owner of the token contract being updated.
+    * The transaction's `sender_address` principal should match the principal contained in the
+      notification's `payload.contract-id`.
+
+Notifications that do not meet these requirements must be ignored.
+
+### Other implications
+
+* Notifications can come at any point in time and are persistent in the Stacks blockchain.
+  * When performing a local sync to the chain tip, old notifications for old metadata updates could
+    not necessarily have a distinct effect in metadata responses when processed in the present.
+* Multiple notifications for the same tokens will not necessarily correspond to multiple metadata
+  updates.
+  * Refreshing a token's metadata should be an idempotent operation. Repeated refreshes should not
+    create distinct records in the internal metadata database.
+  * To prevent slow performance and guard against any Denial of Service attack attempts, contract
+    call rate limiting should be implemented locally.
+* Notifications can be delayed and out of order.
+  * A notification transaction's timestamp should not be considered to be the time when the token
+    metadata was actually updated.
+
+Given these constraints the notifications this SIP proposes should be taken as _hints_ to metadata
+indexers. Metadata indexers are not obliged to follow them.
+
+# Backwards compatibility
+
+Developers who need to emit metadata update notifications for tokens declared in older contracts
+(that were deployed before this notification standard was established) could do so by either calling
+the contract described in [Reference Implementations](#reference-implementations) or by first
+deploying a new separate contract containing a public function that prints this notification and
+then calling it to have it emitted.
+
```

Comment:
I think you're still missing your `Related Work` section.  I think this section should speak to why metadata updates should be sent on-chain, as opposed through some sort of off-chain notification service.

Reply:
Good idea. Done.